openjdk/sun/nio/fs/NetFileSystemProvider.java

```diff
@@ -371,7 +372,40 @@ private static FileDescriptor open(String path, int mode, int rights, int share,
             if (false) throw new cli.System.IO.IOException();
             if (false) throw new cli.System.Security.SecurityException();
             if (false) throw new cli.System.UnauthorizedAccessException();
-            return FileDescriptor.fromStream(new FileStream(path, FileMode.wrap(mode), FileSystemRights.wrap(rights), FileShare.wrap(share), 8, FileOptions.wrap(options)));
+
```

Comment:
I think the old way should be insed an #if NETFRAMEWORK and what is here inside the #else

Reply:
`#if NETFRAMEWORK` doesn't work in Java and I'm somewhat sure that I changed the implementation because things failed at runtime. So, does one need to check at runtime the current version of .NET executed under and deal accordingly? Reads complicated to me:

https://stackoverflow.com/a/51974946/2055163